image_writers_guide/guidelines.adoc

```diff
@@ -9,4 +9,85 @@
 
 toc::[]
 
-== Overview
\ No newline at end of file
+== Overview
+When creating Docker images to run on OpenShift there are a number of best practices to consider in order to ensure consumers of those images have a good experience.  Since images are intended to be immutable and used as is, it is the image author's responsibility to ensure the image contains any artifacts that would typically be needed by a user of the image and that the image can be configured by the user.  Here are some general comments around things to consider.
+
+== General Docker Best Practices
+The following are some guidelines that apply to Docker image creation in general, independent of whether the images are used on OpenShift v3.
+
+=== Reuse images
+Whereever possible you should base your image on an appropriate upstream image via the `FROM` statement.  This will ensure that your image can easily pick up security fixes
+from an upstream image when it is updated, rather than you being responsible for updating your dependencies directly.  
+
+In addition, you should use tags in the `FROM` instruction (for example `rhel:rhel7`) will make it clear to users exactly which version of an image your image is based on.  If you use a tag other than `latest` this ensures that your image is not subjected to breaking changes that might go into the `latest` version of an upstream image.
```

Comment:
Probably a separate topic here on "how to label your image repositories"

We should note - depending on "latest" usually means "I want whatever is at all available".  However, in the long term, we recommend coming up with labels for your image "streams" that corresponds to a commitment to backwards compatibility.  For example, "rhel" comes with the "rhel7" label, which lets people know that they will get all of the updates for rhel7 from that channel, and that RHEL is expecting the backwards compatibility of content built on top of that image will continue on that label.  So we would recommend always creating a separate label to represent what you will provide as backwards compatibility - you might have:

* "php" 
  * "latest" - is whatever the most recent "stable" commit
  * "5.6" - might match latest today, but in the future when "5.7" is introduced will no longer match latest
  * "5.5" - may not be available

Generally, labels should represent stable things that you don't plan to break.  If you do that, then users and administrators can choose when to jump to a new stable version.  It's better to be too specific (major versions and minor versions) than under specific - i.e. if you use "5.x" for PHP, and PHP delivers a non backwards compatible change, applications that depend on 5.x would break.